drivers/hmis/app/models/hmis/auth_policies/README.md

```diff
@@ -0,0 +1,160 @@
+    # Check if assigned to this specific step...
+  end
+end
+```
+
+### 2. **UserContext** (Request-Level Cache)
+- **Location**: `drivers/hmis/app/models/hmis/auth_policies/user_context.rb`
+- **Purpose**: Facade that provides authorization data to policies with heavy memoization
+- **Key Feature**: Context is shared across policies
+
+**What it provides:**
+```ruby
+context.project_permissions(project_id)  # => Set of permissions for a project
+context.assigned_referral_step_ids           # => Set of step IDs user is assigned to
+context.potential_permissions                # => All permissions user could have
+```
+
+### 3. **Context Loaders**
+- **Location**: `drivers/hmis/app/models/hmis/auth_policies/context_loaders/`
+- **Purpose**: Efficient batch loading of authorization data from the database
+- **Examples**:
+  - `HmisProjectAccessGroupLoader` - loads access groups for projects
+  - `CeReferralAssignmentLoader` - loads referral assignments for users
+
+### 4. **Models**
+- **Responsibility**: Declare their policy class via `policy_class` method
+- **Convention**: Keep authorization logic OUT of models - delegate to policies
+
+```ruby
+class Hmis::Hud::Project < Hmis::Hud::Base
+  def self.policy_class = Hmis::AuthPolicies::HmisProjectPolicy
+  def policy_class = self.class.policy_class
+end
+```
+
+### 5. **Consumers** (GraphQL/Controllers)
+- **Location**: GraphQL Resolvers, Controllers, etc.
+- **Usage**: Call `user.policy_for(resource)` then check permissions
+
+## Architecture Flow
+
+```mermaid
+graph TD;
+    subgraph Consumer Layer
+        A[GraphQL Resolver / Controller]
+    end
+
+    subgraph Authorization Framework
+        B["current_user.policy_for(resource)"]
+        C["Policy Class<br>(e.g., CeReferralPolicy)"]
+        D["UserContext<br>(Request-level cache)"]
+        E["Context Loaders<br>(Bulk data loading)"]
+        F[(Database)]
+    end
+
+    A -->|"1 Request authorization"| B;
+    B -->|"2 Instantiate policy"| C;
+    C -->|"3 Query for data"| D;
+    D -->|"4 Load data (cached)"| E;
+    E -->|"5 Fetch from DB (once)"| F;
+    C -->|"6 Return true/false"| A;
+```
+
+## Usage Patterns
+
+### Basic Authorization Check
+```ruby
+# In a GraphQL mutation
+def resolve(referral_id:)
+  referral = Hmis::Ce::Referral.find(referral_id)
+  access_denied! unless policy_for(referral).can_view?
+  # ... continue with business logic
+end
+```
+
+### Class-Level Authorization
+```ruby
+# Check if user can perform any staff assignments
+def staff_assignments
+  return Hmis::StaffAssignment.none unless policy_for(Hmis::StaffAssignment).can_index?
+  # ... return assignments
+end
+```
+
+### Complex Authorization with Context
+```ruby
+# Authorization that depends on multiple factors
+def can_create_referral?(client:)
+  return false unless client.data_source_id == user.hmis_data_source_id
+  return false unless context.project_role_permissions(opportunity.project_id).include?(:can_start_referrals)
```

Comment:
```suggestion
  return false unless context.project_permissions(opportunity.project_id).include?(:can_start_referrals)
```

(It looks like this is the HMIS equivalent of the warehouse project_role_permissions)